### PR TITLE
Zookeeper

### PR DESCRIPTION
**Pull request by scarabus**
An implementation of locking using Zookeeper - integration and fixes of the work begun by Mel.

### CODE REVIEW

File: Apps/PluginInstaller/src/main/java/rapture/plugin/install/PluginShell.java [hunk 1]
```diff
@@ -229,6 +230,8 @@ private void go() {
                 }
                 eval(line);
             }
+        } catch (NoSuchElementException nsee) {
+            System.out.println("Be seeing you");
```
Comment (jonathan-major) [inline Apps/PluginInstaller/src/main/java/rapture/plugin/install/PluginShell.java]: use log statement

File: Libs/RaptureAddins/ZooKeeper/src/main/java/rapture/lock/zookeeper/ZooKeeperLockHandler.java [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+                    logger.error(" connection suspended to zookeeper server");
+                    // TODO we can wait for state to move back to CONNECTED. if not time out with exception
+                }
+            }
+        });
+        client.start();
+    }
+
+    public ZooKeeperLockHandler() {
+        this("localhost:2181", 0);
```
Comment (jonathan-major) [inline Libs/RaptureAddins/ZooKeeper/src/main/java/rapture/lock/zookeeper/ZooKeeperLockHandler.java]: is it always port 2181?

File: Libs/RaptureCore/src/main/java/rapture/kernel/SeriesApiImpl.java [hunk 3]
```diff
@@ -156,8 +156,11 @@ public void deleteSeriesRepo(CallingContext context, String seriesURI) {
     @Override
     public void deletePointsFromSeriesByPointKey(CallingContext context, String seriesURI, List<String> columns) {
         RaptureURI internalURI = new RaptureURI(seriesURI, Scheme.SERIES);
-        SeriesRepo repo = getRepoOrFail(internalURI);
-        repo.deletePointsFromSeriesByColumn(internalURI.getDocPath(), columns);
+        String docPath = internalURI.getDocPath();
```
Comment (jonathan-major) [inline Libs/RaptureCore/src/main/java/rapture/kernel/SeriesApiImpl.java]: is there an associated ticket for this change?

File: Libs/RaptureAddins/ZooKeeper/src/main/java/rapture/lock/zookeeper/ZooKeeperLockHandler.java (hunk 2 above)
Comment (scarabus) [inline Libs/RaptureAddins/ZooKeeper/src/main/java/rapture/lock/zookeeper/ZooKeeperLockHandler.java]: That is the default, but you can specify any host and port by using the other constructor.

File: Libs/RaptureCore/src/main/java/rapture/kernel/SeriesApiImpl.java (hunk 3 above)
Comment (scarabus) [inline Libs/RaptureCore/src/main/java/rapture/kernel/SeriesApiImpl.java]: It's related to RAP-4199 (and RAP-4204)

File: Libs/RaptureAddins/ZooKeeper/src/test/java/rapture/lock/zookeeper/ZooKeeperLockHandlerTest.java [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+        assertEquals(4, handlesDenied.size());
+    }
+
+    private boolean lockExists(String lockName, String lockHolder) {
+        Stat exists;
+        try {
+            exists = client.checkExists().forPath(z.createLockPath(lockName));
+            return (exists != null);
+        } catch (Exception e) {
+            System.out.println(ExceptionToString.format(e));
```
Comment (dukenguyen) [inline Libs/RaptureAddins/ZooKeeper/src/test/java/rapture/lock/zookeeper/ZooKeeperLockHandlerTest.java]: logging

**Pull request merged**